src/map.rs

```diff
@@ -261,29 +261,37 @@ impl<K: Ord, V: Val<A>, A: Actor> Map<K, V, A> {
     }
 
     /// Update a value under some key, if the key is not present in the map,
-    /// the updater will be given the result of V::default().
-    pub fn update<F, I>(&self, key: I, ctx: AddCtx<A>, f: F) -> Op<K, V, A>
+    /// the updater will use the given `default` value.
+    pub fn update<F>(&self, key: K, ctx: AddCtx<A>, default: V, f: F) -> Op<K, V, A>
```

Comment:
shouldn't this one get the "update_with_default" name, cause it's called with an default? it would more accurately match rust conventions (eg `Vec::with_capacity`). also it would be less of a breaking change.

Reply:
Ya, I debated this in my head. I was thinking the `with_default` would refer to the trait. But, the example of `with_capacity` makes sense and backwards compat :+1: I'll update this :)

Reply:
Updated :ok_hand: